application-templates/starter-typescript/src/components/channel-details/channel-details-form.tsx

```diff
@@ -0,0 +1,111 @@
+import type { ReactElement } from 'react';
+import { useFormik, type FormikHelpers } from 'formik';
+import { useIntl } from 'react-intl';
+import LocalizedTextField from '@commercetools-uikit/localized-text-field';
+import TextField from '@commercetools-uikit/text-field';
+import Spacings from '@commercetools-uikit/spacings';
+import SelectField from '@commercetools-uikit/select-field';
+import type { TFormValues } from '../../types';
+import { CHANNEL_ROLES } from './constants';
+import validate from './validate';
+import messages from './messages';
+
+type TChannelRole = keyof typeof CHANNEL_ROLES;
+const getRoleOptions = Object.keys(CHANNEL_ROLES).map((key) => ({
+  label: CHANNEL_ROLES[key as TChannelRole],
+  value: CHANNEL_ROLES[key as TChannelRole],
+}));
+
+type Formik = ReturnType<typeof useFormik>;
+type FormProps = {
+  formElements: ReactElement;
+  values: Formik['values'];
+  isDirty: Formik['dirty'];
+  isSubmitting: Formik['isSubmitting'];
+  submitForm: Formik['handleSubmit'];
+  handleReset: Formik['handleReset'];
+};
+
+type ChannelDetailsFormProps = {
+  onSubmit: (
+    values: TFormValues,
+    formikHelpers: FormikHelpers<TFormValues>
+  ) => void | Promise<unknown>;
+  initialValues: TFormValues;
+  isReadOnly: boolean;
+  dataLocale: string;
+  children: (formProps: FormProps) => JSX.Element;
+};
+
+const ChannelDetailsForm = (props: ChannelDetailsFormProps) => {
+  const intl = useIntl();
+  const formik = useFormik<TFormValues>({
+    initialValues: props.initialValues,
+    onSubmit: props.onSubmit,
+    validate,
+    enableReinitialize: true,
+  });
+
+  const formElements = (
+    <Spacings.Stack scale="l">
+      <TextField
+        name="key"
+        title={intl.formatMessage(messages.channelKeyLabel)}
+        value={formik.values.key}
+        errors={TextField.toFieldErrors<TFormValues>(formik.errors).key}
+        touched={formik.touched.key}
+        onChange={formik.handleChange}
+        onBlur={formik.handleBlur}
+        isReadOnly={props.isReadOnly}
+        renderError={(errorKey) => {
+          switch (errorKey) {
+            case 'duplicate':
+              return intl.formatMessage(messages.duplicateKey);
+            default:
+              return null;
+          }
```

Comment:
NIT: Since we only have two options here, I would prefer an If statement. 

```suggestion
          if (errorKey === 'duplicate') {
              return intl.formatMessage(messages.duplicateKey);
          }
          return null;
```